subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheEncryptionIntegrationTest.groovy

```diff
@@ -0,0 +1,251 @@
+            "SENSITIVE_ENV_VAR_NAME": 'sensitive_env_var_value'
+        ])
+
+        then:
+        configurationCache.assertStateStored()
+        def cacheDir = new File(this.testDirectory, ".gradle/configuration-cache")
+        isFoundInDirectory(cacheDir, "sensitive_property_name".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_property_value".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_property_name2".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_property_value2".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_value1".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_value2".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_env_var_value".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_value3".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive_convention".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "sensitive".getBytes()) == !enabled
+        isFoundInDirectory(cacheDir, "SENSITIVE".getBytes()) == !enabled
+        where:
+        encrypted       |   enabled
+        "unencrypted"   |   false
+        "encrypted"     |   true
+    }
+
+    private boolean isFoundInDirectory(File startDir, byte[] toFind) {
+        try (Stream<Path> tree = Files.walk(startDir.toPath(), FileVisitOption.FOLLOW_LINKS)) {
+            return tree.filter { it.toFile().file }
+                .anyMatch {
+                    isSubArray(Files.readAllBytes(it), toFind)
+                }
+        }
+    }
+
+    def "new configuration cache entry if was encrypted but encryption is off"() {
+        given:
+        def configurationCache = newConfigurationCacheFixture()
+        runWithEncryption(true)
+
+        when:
+        runWithEncryption(false)
+
+        then:
+        configurationCache.assertStateStored()
+        outputContains("Calculating task graph as no configuration cache is available for tasks: help")
+    }
+
+    def "new configuration cache entry if was not encrypted but encryption is on"() {
+        given:
+        def configurationCache = newConfigurationCacheFixture()
+        runWithEncryption(false)
+
+        when:
+        runWithEncryption(true)
+
+        then:
+        configurationCache.assertStateStored()
+        outputContains("Calculating task graph as no configuration cache is available for tasks: help")
+    }
+
+    def "encryption disabled if requested but key in env var is not present"() {
+        given:
+        def configurationCache = newConfigurationCacheFixture()
+
+        when:
+        runWithEncryption(true, ["help"], [], [(GRADLE_ENCRYPTION_KEY_ENV_KEY): ""])
+
+        then:
+        configurationCache.assertStateStored()
+        outputContains("Encryption was requested but could not be enabled")
```

Comment:
🔧 This condition should be reported as a problem (and fail the build unless `o.g.c-c.problems=warn`).

Reply:
That would force users to specify a valid key whenever CC is enabled (or explicitly disable encryption). I thought having a warning would be middle ground.

Reply:
What about a warning plus causing the cache to be discarded?